Lib/timeSeries.py

```diff
@@ -84,4 +101,317 @@ def readUnrTxyz2(self, fileName):
+
+                # set coordinate type to differential coordinates
+                self.coordType = 'dXdYdZ'
+
+            else:
+
+                # raise coordTransformError
+                msg = ("ERROR: timeSeries.coordType must equal 'XYZ' to " + 
+                       "perform timeSeries.setRefPosToAvg()")
+                raise CoordTransformError(msg)
+
+        except CoordTransformError as cte:
+
+            print(cte)
+
+    ####################################################################
+    def dxdydz2enu(self):
+
+        """
+        Transform coordinates from dXdYdZ to local East, North, Up 
+        coordinate system. Automatically changes timeSeries.coordType
+        to 'ENU' and timeSeries.refPos to units of Lon., Lat., and Ht.
+        """
+
+        # this routine should only be used when the coordinates have
+        # been reference to a local point, like what is done using
+        # timeSeries.setRefPosToAvg(). coordType should be 'dXdYdZ'
+        try:
+            if self.coordType == 'dXdYdZ':
+                
+                # set new coorType to 'ENU'
+                self.coordType = 'ENU'
+                
+                ### 
+                # transform refPos from XYZ -> Lon, Lat, Ht
+                ###
+                refPosX = self.refPos[0]
+                refPosY = self.refPos[1]
+                refPosZ = self.refPos[2]
+
+                # transform.xyz_to_llh returns: lat, lon, ht
+                # so switch first two returns to make refPos
+                # lon, lat, ht
+                refPosLLH = np.asarray([0.,0.,0.])
+                refPosLLH[1], refPosLLH[0], refPosLLH[2] = transform.xyz_to_llh(
+                                           refPosX, refPosY, refPosZ)
+
+                self.refPos = refPosLLH
+    
+                ###
+                # convert coordinates to ENU using transform.xyz_to_enu()
+                ###
+                inPos = self.pos
+
+                # loop over the columns of inPos
+                for i in range(0, np.shape(inPos)[1]):
+
+                    dX = inPos[0][i]
+                    dY = inPos[1][i]
+                    dZ = inPos[2][i]
+                    
+                    # again transform.xyz_to_enu wants lat first 
+                    # then lon so switch the order of refPos[0]
+                    # and refPos[1]
+                    E, N, U = transform.xyz_to_enu( self.refPos[1],
+                                self.refPos[0], dX, dY, dZ)
+
+                    self.pos[0][i] = E
+                    self.pos[1][i] = N
+                    self.pos[2][i] = U
+
+                ###
+                # convert var/covar mtx to ENU using 
+                # transform.xyz_to_enu_cov()
+                ###
+                inSig = self.sig
+                inCorr = self.corr
+
+                # loop over the columns of inSig and inCovar 
+                # (should be same size)
```

Comment:
Should you add error handling for when they aren't the case?

Reply:
Yes, but not here. When the .txyz2 file is read in it should check to make sure that all of the arrays that *should* be the same size *are* the same size.